File: pegasusX/apps/supplier-app-android/app/src/main/java/com/pegasusx/supplier/ui/screens/orgfleet/OrgFleetUtils.kt
package com.pegasusx.supplier.ui.screens.orgfleet

import com.pegasusx.supplier.data.model.SupplierTopologyResponse

fun nodeLabel(type: String, id: String, topology: SupplierTopologyResponse?): String {
    if (topology == null) return id
    if (type == "FACTORY") {
        return topology.factories.find { it.factoryId == id }?.name ?: id
    }
    return topology.warehouses.find { it.warehouseId == id }?.name ?: id
}

File: pegasusX/apps/supplier-app-ios/SupplierApp/Views/OrgFleet/Components/OrgFleetUtils.swift
import Foundation

func nodeLabel(topology: SupplierTopologyResponse?, type: String, id: String) -> String {
    guard let topology else { return id }
    if type == "FACTORY" {
        return topology.factories.first { $0.factoryId == id }?.name ?? id
    }
    return topology.warehouses.first { $0.warehouseId == id }?.name ?? id
}
